src/wallet/wallet_rpc_server.cpp

```diff
@@ -2444,6 +2444,28 @@ namespace tools
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_refresh(const wallet_rpc::COMMAND_RPC_REFRESH::request& req, wallet_rpc::COMMAND_RPC_REFRESH::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_wallet) return not_open(er);
+    if (m_wallet->restricted())
+    {
+      er.code = WALLET_RPC_ERROR_CODE_DENIED;
+      er.message = "Command unavailable in restricted mode.";
+      return false;
+    }
+    try
+    {
+      m_wallet->refresh(req.trust_daemon, req.start_height, res.blocks_fetched, res.received_money);
```

Comment:
The setting for trusting the daemon or not is already held by the wallet instance, in this case `m_trusted_daemon`, so it should be used here.